### PR TITLE
Add clarity/retries to proxy url test

### PR DESCRIPTION
**Pull request by bprashanth**
Improve one segment of the kube-proxy networking test by:      
1. Retrying for 30s 
2. Bucketing into 2 failure modes
3. Adding some clarity by describing the exec pod on failure

Althought 1 shouldn't be necessary, I don't think we lose anything if the kube-proxy convenience endpoint doesn't respond immediately, and if it fails for 30s straight it is indicative of something that requires attention probably within 1.5. 

Fixes https://github.com/kubernetes/kubernetes/issues/32436

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36918)

### CODE REVIEW

**Comment by bprashanth** [comment]
P2 because it addresses an ongoing flake

File: test/e2e/framework/networking_utils.go [hunk 1]
```diff
@@ -269,8 +269,33 @@ func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targ
 func (config *NetworkingTestConfig) GetSelfURL(path string, expected string) {
 	cmd := fmt.Sprintf("curl -q -s --connect-timeout 1 http://localhost:10249%s", path)
 	By(fmt.Sprintf("Getting kube-proxy self URL %s", path))
-	stdout := RunHostCmdOrDie(config.Namespace, config.HostTestContainerPod.Name, cmd)
-	Expect(strings.Contains(stdout, expected)).To(BeTrue())
+
+	// These are arbitrary timeouts. The curl command should pass on first try,
+	// unless kubeproxy is starved/bootstrapping/restarting etc.
+	interval := 1 * time.Second
```
Comment (bowei) [inline test/e2e/framework/networking_utils.go]: const

File: test/e2e/framework/networking_utils.go [hunk 2]
```diff
@@ -269,8 +269,33 @@ func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targ
...
+	timeout := 30 * time.Second
+	podName := config.HostTestContainerPod.Name
+	var msg string
+	if pollErr := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		stdout, err := RunHostCmd(config.Namespace, podName, cmd)
+		if err != nil {
+			msg = fmt.Sprintf("failed executing cmd %v in %v/%v: %v", cmd, config.Namespace, podName, err)
+			Logf(msg)
```
Comment (bowei) [inline test/e2e/framework/networking_utils.go]: doesn't Logf do formatting as part of the function?

Logf(""failed executing cmd %v in %v/%v: %v", cmd, ...)

File: test/e2e/framework/networking_utils.go [hunk 3]
```diff
@@ -269,8 +269,33 @@ func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targ
...
+			return false, nil
+		}
+		if !strings.Contains(stdout, expected) {
+			msg = fmt.Sprintf("successfully executed %v in %v/%v, but output '%v' doesn't contain expected string '%v'", cmd, config.Namespace, podName, stdout, expected)
+			Logf(msg)
```
Comment (bowei) [inline test/e2e/framework/networking_utils.go]: see above

File: test/e2e/framework/networking_utils.go [hunk 4]
```diff
@@ -269,8 +269,33 @@ func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targ
...
+			return false, nil
+		}
+		return true, nil
+	}); pollErr != nil {
+		Logf("\nOutput of kubectl describe pod %v/%v:\n", config.Namespace, podName)
+		desc, _ := RunKubectl(
+			"describe", "pod", podName, fmt.Sprintf("--namespace=%v", config.Namespace))
+		Logf(desc)
```
Comment (bowei) [inline test/e2e/framework/networking_utils.go]: probably should do Logf("%s", desc), if desc contains an embedded %

**Review by bowei — COMMENTED** [review]
nits, otherwise lgtm

File: test/e2e/framework/networking_utils.go (hunk 2 above)
Comment (bprashanth) [inline test/e2e/framework/networking_utils.go]: i'm saving it in msg to reuse in the failure message

**Comment by bprashanth** [comment]
PTAL

**Review by bowei — APPROVED** [review]
/lgtm

**Pull request merged**